src/attendance/attendance.controller.ts

```diff
@@ -1,13 +1,130 @@
-import { Controller, Get, Post, Put, Patch, Param } from '@nestjs/common';
-import { AttendanceService } from '../adapters/default/attendance.adapter';
-
-@Controller('attendance')
+import { AttendanceService } from "../adapters/default/attendance.adapter";
+import {
+  ApiTags,
+  ApiBody,
+  ApiOkResponse,
+  ApiForbiddenResponse,
+  ApiCreatedResponse,
+  ApiBasicAuth,
+  ApiConsumes,
+} from "@nestjs/swagger";
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Put,
+  Patch,
+  Param,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  SerializeOptions,
+  Req,
+  Request,
+  UploadedFile,
+} from "@nestjs/common";
+import { AttendanceDto } from "./dto/attendance.dto";
+import { request } from "http";
+import { FileInterceptor } from "@nestjs/platform-express";
+import { diskStorage } from "multer";
+import { editFileName, imageFileFilter } from "./utils/file-upload.utils";
+import { AttendanceSearchDto } from "./dto/attendance-search.dto";
+@ApiTags("Attendance")
+@Controller("attendance")
 export class AttendanceController {
+  constructor(private service: AttendanceService) {}
+
+  @UseInterceptors(ClassSerializerInterceptor)
+  @Get("/:id")
+  @ApiBasicAuth("access-token")
+  @ApiCreatedResponse({ description: "Attendance detail" })
+  @ApiForbiddenResponse({ description: "Forbidden" })
+  @SerializeOptions({
+    strategy: "excludeAll",
+  })
+  public async getAttendance(
+    @Param("id") attendanceId: string,
+    @Req() request: Request
+  ) {
+    return this.service.getAttendance(attendanceId, request);
+  }
+
+  @Post()
+  @ApiConsumes("multipart/form-data")
```

Comment:
Am merging this for now, so integration can begin. However, the implementation might change, based on the outcome of this discussion https://github.com/shiksha-platform/shiksha-community/discussions/20